_includes/https_redirect.html

```diff
@@ -0,0 +1,8 @@
+<script>
+  // If we're running on a real web server (as opposed to localhost on a custom port,
+  // which is whitelisted), then change the protocol to HTTPS.
+  // See https://goo.gl/lq4gCo for an explanation.
+  if ((!location.port || location.port == "80") && location.protocol != 'https:') {
```

Comment:
I borrowed this logic from https://github.com/jakearchibald/isserviceworkerready/blob/master/src/base.html#L8

I think the idea is that checking for `location.origin !== 'localhost'` is overly restrictive, since there are IP4/IP6 addresses that are also whitelisted and would work.

The existing logic is a little hard to follow, but it does allow for use of localhost with any port other than 80.